tests/components/rfxtrx/test_binary_sensor.py

```diff
@@ -50,15 +52,36 @@ async def test_one_pt2262(hass, rfxtrx):
     assert state.state == "off"  # probably aught to be unknown
     assert state.attributes.get("friendly_name") == "PT2262 22670e"
 
-    await _signal_event(hass, "0913000022670e013970")
+    await rfxtrx.signal("0913000022670e013970")
     state = hass.states.get("binary_sensor.pt2262_22670e")
     assert state.state == "on"
 
-    await _signal_event(hass, "09130000226707013d70")
+    await rfxtrx.signal("09130000226707013d70")
     state = hass.states.get("binary_sensor.pt2262_22670e")
     assert state.state == "off"
 
 
+@pytest.mark.parametrize(
+    "state,event",
+    [["on", "0b1100cd0213c7f230010f71"], ["off", "0b1100cd0213c7f230000f71"]],
+)
+async def test_state_restore(hass, rfxtrx, state, event):
+    """State restoration."""
+
+    entity_id = "binary_sensor.ac_213c7f2_48"
+
+    mock_restore_cache(hass, [State(entity_id, state, attributes={ATTR_EVENT: event})])
+
+    assert await async_setup_component(
+        hass,
+        "rfxtrx",
+        {"rfxtrx": {"device": "abcd", "devices": {"0b1100cd0213c7f230010f71": {}}}},
+    )
+    await hass.async_block_till_done()
+
+    assert hass.states.get(entity_id).state == state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)